gpsd_client/src/client.cpp

```diff
@@ -18,7 +21,11 @@ namespace gpsd_client
       use_gps_time_(true),
       check_fix_by_variance_(true),
       frame_id_("gps")
-    {}
+    {
+      timer_ = create_wall_timer(1s, std::bind(&GPSDClientComponent::step, this));
```

Comment:
While most GPS devices publish data at 1 Hz, this should be possible to adjust with a parameter. I would suggest adding:

```suggestion
      double gps_hz;
      get_parameter_or("gps_hz_hint", gps_hz, 1.0);
      
      // We want to call gps_waiting at a higher rate than how often data is actually published to avoid
      // delays. Otherwise we may always be reading the data at the end of the 1 hz period.
      // Hence multiplying by 2 here.
      int gps_period = static_cast<int>(1000 / gps_hz) * 2;
      
      timer_ = create_wall_timer(
        std::chrono::milliseconds(gps_period), 
        std::bind(&GPSDClientComponent::step, 
        this)
      );
```